prow/hook/server.go

```diff
@@ -53,7 +58,23 @@ type Server struct {
 
 // ServeHTTP validates an incoming webhook and puts it into the event channel.
 func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	eventType, eventGUID, payload, ok, resp := github.ValidateWebhook(w, r, s.TokenGenerator)
+	tokenSecret, revision := s.TokenGenerator()
+	if revision != atomic.LoadUint32(&s.currentTokenRevision) {
+		atomic.StoreUint32(&s.currentTokenRevision, revision)
```

Comment:
I don't think  we should store the revision before we updated the map, especially since the unmarshaling might fail. IMHO we should simply use a `sync.RWMutex` here require the write lock for the whole update section

Reply:
The main purpose of this PR is to reduce the frequency of unmarshalling the secret string into a map - only do unmarshall when the secret changes rather than in every webhook request.

If we only use a `sync.RWMutex` but do not store the revision, we won't be able to know when the secret is changed thus cannot implement what we want above. Or are you having other ideas in mind?

Reply:
I didn't mean that we should not store the revision, I meant that we should only update the revision after we have updated the value. Right now it may happen that we bump the revision and then fail or that concurrent requests will fail because we bumped the revision already but didn't yet bump the value

Reply:
Ah I see, that is a bug... I have made the change and switched to `sync.RWMutex`.

Thanks!